sdk/eventhub/azure-eventhub/azure/eventhub/aio/_transport/_pyamqp_transport_async.py

```diff
@@ -161,6 +161,7 @@ async def _callback_task(consumer, batch, max_batch_size, max_wait_time):
             events = [EventData._from_message(message) for message in messages] # pylint: disable=protected-access
             now_time = time.time()
             if len(events) > 0:
+                consumer._last_received_event = events[-1] # pylint: disable=protected-access
                 await consumer._on_event_received(events if batch else events[0]) # pylint: disable=protected-access
                 consumer._last_callback_called_time = now_time # pylint: disable=protected-access
             else:
```

Comment:
Line 199 in `receive_task` is what was being impacted by this as on error consumer._last_received_event was always none